pkg/capabilities/registry/basic.go

```diff
@@ -0,0 +1,141 @@
+package registry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+)
+
+var (
+	ErrCapabilityAlreadyExists = errors.New("capability already exists")
+)
+
+type Basic struct {
```

Comment:
This can be private, right?

Also let's not call it "Basic". Maybe just "Registry" like in core?

Reply:
registry.Registry is discouraged by Go's naming convention. I'm open to other suggestions.

Reply:
sigh... CapabilitiesRegistry? :)